app/javascript/helpers/forms/EditRoleFormHelpers.jsx

```diff
@@ -1,18 +1,28 @@
 import * as yup from 'yup';
 import { yupResolver } from '@hookform/resolvers/yup';
 
-const validationSchema = yup.object({
+const validationSchemaRoleName = yup.object({
   // TODO: amir - Revisit validations.
   name: yup.string().required('Please enter the role name.'),
 });
 
-export const editRoleFormConfig = {
+const validationSchemaRoomLimit = yup.object({
+  roomLimit: yup.number().required('Please enter the room count limit.').min(0, 'Minimum atleast 0').max(100, 'Allowed maximum is 100'),
+});
+
+export const editRoleFormConfigRoleName = {
+  mode: 'onBlur',
+  defaultValues: {},
```

Comment:
It's recommended to provide in `defaultValues` even if we have a `useEffect` hook that will set values,
The form will render before the hook will be called,

Reply:
Here that is causing the problem where you blur the input without typing any value you'll get this weird error message:
`roomLimit must be a `number` type, but the final value was: `NaN` (cast from the value `""`).`
![image](https://user-images.githubusercontent.com/29759616/194072569-c0a12cfc-4070-4ac1-9bc0-ab6dc99ed9cf.png)

With the issues noted above that undefined value will be passed to the API:
![image](https://user-images.githubusercontent.com/29759616/194072452-c877b69e-7386-4375-a80b-f107c16a8983.png)